### PR TITLE
feat(extensions): expose ExtensionRegistryObserver events in Session

### PR DESCRIPTION
**Pull request by samuelmaddock**
#### Description of Change
Extensions can be loaded or unloaded for various reasons. In some cases, this can occur by no means of the Electron programmer, such as in the case of [`chrome.runtime.reload()`](https://developer.chrome.com/extensions/runtime#method-reload).

In order to be able to manage state about extensions outside of Electron's APIs, events related to loading and unloaded extensions are needed. The only alternative would be to monkey-patch the `Session` prototype.

The use case I have is to manage state in my own implementation of [`chrome.browserAction`](https://developer.chrome.com/extensions/browserAction). I need to know when an extension is added or removed to set or delete saved state.

Ref #19447

cc @nornagon @sentialx 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Add `extension-loaded`, `extension-unloaded`, and `extension-ready` events to `Session`.

### CODE REVIEW

**Comment by samuelmaddock** [comment]
Pinging @electron/wg-api since I think this will require their review.

File: docs/api/session.md
Comment (nornagon) [inline docs/api/session.md:110]: Perhaps include this in the visible part of the documentation?

Or, better, fully enumerate the situations that can cause this event in Electron.

File: docs/api/session.md
Comment (nornagon) [inline docs/api/session.md:119]: When does this happen in Electron?

File: shell/browser/api/electron_api_session.cc
Comment (nornagon) [inline shell/browser/api/electron_api_session.cc:303]: I think this is handled by `ScopedObserver`, yes?

Given we're only ever only ever observing a single source, perhaps the scoped observer isn't needed?

**Review by nornagon — CHANGES_REQUESTED** [review]
lgtm, just a few minor comments

File: shell/browser/api/electron_api_session.cc
Comment (samuelmaddock) [reply shell/browser/api/electron_api_session.cc:303]: The `ScopedObserver` wasn't really working how I thought it did so I just removed it. It would need to be passed the `browser_context()` which the template doesn't support.

File: docs/api/session.md
Comment (samuelmaddock) [reply docs/api/session.md]: I've updated it to include more info, but the parts about enabled/disabled extensions doesn't make a lot of sense until #25386 is merged. However, enabled/disabled extensions still are technically part of the extension system implementation regardless.

File: docs/api/session.md
Comment (samuelmaddock) [reply docs/api/session.md]: Updated to mention `Session.removeExtension`

File: docs/api/session.md
Comment (samuelmaddock) [reply docs/api/session.md]: I'll hold off on mentioning the concept of enabled/disabled extensions for now. I'll plan to include them in the mentioned PR.

**Review by nornagon — APPROVED** [review]

**Comment by jkleinsc** [comment]
The API WG approved this at the Sept 21, 2020 meeting.

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**
